### PR TITLE
FIX: Fix many examples

### PR DESCRIPTION
**Pull request by Eric89GXL**
One example updated and I plan to add more.

Addresses https://github.com/vispy/vispy/issues/965

~~While I was updating it @campagnola I noticed the mouse interactivity for `panzoom` does not work. The mouse callback is not called once I add the visual -- if the visual is not there, the mouse callback is called (I put in a debugging statement at one point to check). Any ideas?~~

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Never mind that last question, I got it working. On to benchmark 2.

**Comment by Eric89GXL** [comment]
Okay @campagnola, tutorials translated. I tweaked `visual` a bit to have some sensible defaults for `_prepare_draw` (do nothing) and `_prepare_transforms` (set `view.view_program.vert['transform']`) because it seemed to save on lines all in all. But if you'd prefer to have some repeated lines to force people to think about all of those methods, I can the repeated code in the correct places. Let me know what you think.

I did run into one bug with PanZoom where the right-click drag (scale) does not update the plot on my system, even though the left-click drag (translate) does. Do you have that problem, too?

File: examples/tutorial/app/shared_context.py [hunk 1]
```diff
@@ -54,9 +56,11 @@ def __init__(self):
...
 
     def on_init(self, event):
-        self.text = Text('Initialized', font_size=40.,
-                         anchor_x='left', anchor_y='top',
-                         parent=[self.vb_0.scene, self.vb_1.scene])
+        self.text = Text('Initialized', font_size=10.,
+                         anchor_x='left', anchor_y='top')
+        self.views = [self.text.view(), self.text.view()]
+        self.vb_0.add(self.views[0])
+        self.vb_1.add(self.views[1])
```
Comment (Eric89GXL) [inline examples/tutorial/app/shared_context.py]: @campagnola this does not work, and I tried a few things. What is the proper syntax? There are also rendering issues even with a single `vb_0.add(self.text)`, but that might be harder to fix.

**Comment by campagnola** [comment]
> I tweaked visual a bit to have some sensible defaults for _prepare_draw (do nothing) and _prepare_transforms

Good! That was starting to irritate me.

> I did run into one bug with PanZoom where the right-click drag (scale) does not update the plot on my system, even though the left-click drag (translate) does. Do you have that problem, too?

I have not had that problem. Does the wheel work for you? If you want to debug, the path is:

* `SceneCanvas._process_mouse_events()` -- make sure the event is being delivered to the correct visual (viewbox) here: `getattr(picked.events, event.type)(scene_event)`
* `PanZoomCamera.viewbox_mouse_event()`

File: examples/tutorial/app/shared_context.py (hunk 1 above)
Comment (campagnola) [inline examples/tutorial/app/shared_context.py]: The scenegraph does not support views yet; let's leave this one for another PR.

Comment (campagnola) [inline examples/tutorial/app/shared_context.py]: By "rendering issues", do you mean that sometimes the characters are rendered as solid rectangles?

**Comment by Eric89GXL** [comment]
Got rid of the changes to `shared_context.py`. Do you want to merge this as-is for now, and I'll continue tweaking examples in another PR?

BTW more issues with the events system -- `examples/basics/visuals/line.py` runs fine on `master` but fails to update the line dynamically on `scenegraph-update`. I also have an issue where mouse-move events are no longer picked up properly / show up in e.g. `examples/demo/scene/magnify.py`. I'll see if I can debug it.

**Comment by Eric89GXL** [comment]
Regarding rendering issues -- yes, they render as solid rectangles sometimes

**Comment by Eric89GXL** [comment]
Ahh, the problem is the `_send_hover_events` thing. I agree the API needs some thought. On the one hand using `visual_at` is perhaps the most general, but it comes at a high speed cost. I wonder if we can by default have a `'widget'` mode that uses knowledge of the `SceneCanvas` child widget locations to pass a mouse event to the widget directly. It seems like this would make things usable in most cases.

**Comment by Eric89GXL** [comment]
Actually, don't bother merging. I think I'll be able to get `make examples` working in this PR today.

**Comment by Eric89GXL** [comment]
@bollu I took care of IsoLine with this PR, I think ColorBar is the only one left. Let me know if you start working on it, otherwise I might get to it.

**Comment by Eric89GXL** [comment]
@bollu also please look at my changes to make sure they make sense based on the other changes you have made

**Comment by Eric89GXL** [comment]
@campagnola BTW I think there is a problem with `Volume`, try running the examples that use it. We get warnings that the uniforms that the `visual_at` filters use do not exist, suggesting that that visual is not formatted properly. (The properties should be there.)

**Comment by kmuehlbauer** [comment]
@Eric89GXL, please consider review of #12, regarding Isoline/Isocurve. Thanks!

**Comment by Eric89GXL** [comment]
@kmuehlbauer I removed my changes related to isoline/isocurve, so we hopefully won't have any conflicts.

I think to avoid other conflicts I will stop here for now, ready for review/merge @campagnola.

File: examples/basics/scene/magnify.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 # Make a canvas and partition it into 3 viewboxes.
 #
 canvas = vispy.scene.SceneCanvas(keys='interactive', show=True)
+canvas._send_hover_events = True  # temporary workaround
```
Comment (campagnola) [inline examples/basics/scene/magnify.py]: Ha. I added `_send_hover_events` specifically to support this example and never set the flag.

File: examples/basics/visuals/arcball.py [hunk 3]
```diff
@@ -22,21 +22,19 @@ def __init__(self):
...
-        self.tr_sys.visual_to_document = self.cube_transform
+        self.quaternion = Quaternion()
         self.show()
 
     def on_draw(self, event):
-        gloo.set_viewport(0, 0, *self.physical_size)
-        gloo.clear('white')
-        self.tr_sys.auto_configure()
-        self.cube.draw(self.tr_sys)
+        vp = (0, 0, self.physical_size[0], self.physical_size[1])
```
Comment (campagnola) [inline examples/basics/visuals/arcball.py]: These three lines should be in `on_resize()`

File: vispy/scene/widgets/grid.py [hunk 4]
```diff
@@ -200,8 +201,9 @@ def _update_child_widgets(self):
                                                      [cspan, rspan])
         row_occ = occupied.sum(axis=1) > 0
         col_occ = occupied.sum(axis=0) > 0
-        row_stretch = np.nanmean(stretch[..., 1], axis=1)
-        col_stretch = np.nanmean(stretch[..., 0], axis=0)
+        with warnings.catch_warnings(record=True):  # mean of empty slice
```
Comment (campagnola) [inline vispy/scene/widgets/grid.py]: AH. I was wondering where those came from.

**Comment by Eric89GXL** [comment]
Fixed the arcball example as requested

**Comment by campagnola** [comment]
Awesome!
